pkg/step/feature/derivation.go

```diff
@@ -486,6 +487,23 @@ func setBackToIR(trainStmt *ir.TrainStmt, fcMap ColumnMap, columnTargets []strin
 	}
 }
 
+// setBackToLabel update label field meta
+func setBackToLabel(trainStmt *ir.TrainStmt, fmMap FieldDescMap) {
+	// NOTE: clustering model may not specify Label
+	labelName := trainStmt.Label.GetFieldDesc()[0].Name
+	if labelName == "" {
+		return
+	}
+	trainStmt.Label = &ir.NumericColumn{
+		FieldDesc: fmMap[labelName],
+	}
+	// use shape [] if label shape is [1] for Tensorflow scalar label shape should be [].
+	shape := trainStmt.Label.GetFieldDesc()[0].Shape
```

Comment:
Here I don't understand -- how comes shape? And why do we need to set the field Shape?

Reply:
I don't know why Tensorflow only accept shape `[]` for the label, pass `[1]` will cause an error. I'm wondering if we should pass `[None, 1]` because Tensorflow uses `None` to indicate the batch size dimension. I still need to do more test to confirm this.